src/server.ts

```diff
@@ -1,97 +1,246 @@
+import apm from 'elastic-apm-node';
+import { ObjectID } from 'mongodb';
+import { GrpcHealthCheck, HealthCheckResponse, HealthService } from 'grpc-ts-health-check';
+import { FileService } from './file/file.service';
+import { log, Severity } from './utils/logger';
+import { IFile } from './file/file.interface';
+import { apmURL, verifyServerCert, serviceName, secretToken } from './config';
+import { statusToString, validateGrpcError } from './utils/errors/grpc.status';
+import { ApplicationError } from './utils/errors/application.error';
+import { IUpload } from './upload/upload.interface';
+
+apm.start({
+  serviceName,
+  secretToken,
+  verifyServerCert,
+  serverUrl: apmURL,
+});
+
+const PROTO_PATH: string = `${__dirname}/../proto/file.proto`;
+
+// Suggested options for similarity to existing grpc.load behavior
+const packageDefinition : protoLoader.PackageDefinition = protoLoader.loadSync(
+  PROTO_PATH,
+  {
+    keepCase: true,
+    longs: String,
+    enums: String,
+    defaults: true,
+    oneofs: true,
+  });
+
+// Has the full package hierarchy
+const protoDescriptor : grpc.GrpcObject = grpc.loadPackageDefinition(packageDefinition);
+const file_proto : any = protoDescriptor.file;
+
+export const serviceNames: string[] = ['', 'file.fileService'];
+export const healthCheckStatusMap = {
+  '': HealthCheckResponse.ServingStatus.UNKNOWN,
+  serviceName: HealthCheckResponse.ServingStatus.UNKNOWN
+};
+
+/**
+ * The FileServer class, containing all of the FileServer methods.
+ */
+export class FileServer {
+
+  public server: grpc.Server;
+  public grpcHealthCheck: GrpcHealthCheck;
+
+  public constructor(port: string) {
+    this.server = new grpc.Server();
+    this.addServices();
+    this.server.bind(`0.0.0.0:${port}`, grpc.ServerCredentials.createInsecure());
+    log(Severity.INFO, 'server bind', `server listening on port: ${port}`);
+  }
+
+  private addServices () {
+    // Register the health service
+    this.grpcHealthCheck = new GrpcHealthCheck(healthCheckStatusMap);
+    this.server.addService(HealthService, this.grpcHealthCheck);
+
+    const fileService = {
+      GenerateKey: this.wrapper(this.GenerateKey),
+      CreateUpload: this.wrapper(this.CreateUpload),
+      UpdateUploadID: this.wrapper(this.UpdateUploadID),
+      GetUploadByID: this.wrapper(this.GetUploadByID),
+      DeleteUploadByID: this.wrapper(this.DeleteUploadByID),
+      GetFileByID: this.wrapper(this.GetFileByID),
+      GetFileByKey: this.wrapper(this.GetFileByKey),
+      GetFilesByFolder: this.wrapper(this.GetFilesByFolder),
+      CreateFile: this.wrapper(this.CreateFile),
+      DeleteFile: this.wrapper(this.DeleteFile),
+      IsAllowed: this.wrapper(this.IsAllowed),
+    };
+
+    this.server.addService(file_proto.FileService.service, fileService);
+  }
+  /**
+   * wraps all of the service methods, creating the transaction for the apm and the logger.
+   * @param func - the method called.
+   */
+  private wrapper (func: Function) :
```

Comment:
Maybe you shlod call that `wrapWithLog` or somthing like that

Reply:
I think it makes the code look cumbersome. Instead of changing the name, I added documentation.